### PR TITLE
DDCE-5000 Remove pureconfig

### PR DESCRIPTION
**Pull request by gastonracui**

### CODE REVIEW

File: app/uk/gov/hmrc/merchandiseinbaggage/config/AppConfig.scala [hunk 1]
```diff
@@ -16,51 +16,44 @@
 
 package uk.gov.hmrc.merchandiseinbaggage.config
 
+import com.google.inject.Inject
+import play.api.Configuration
+import uk.gov.hmrc.merchandiseinbaggage.config.AppConfig._
 import javax.inject.Singleton
```
Comment (Taiwo-AB) [inline app/uk/gov/hmrc/merchandiseinbaggage/config/AppConfig.scala:22]: Can just use this instead for Inject: `import javax.inject.{Inject, Singleton}` and delete the one on line 19, thanks.

File: app/uk/gov/hmrc/merchandiseinbaggage/config/AppConfig.scala [hunk 2]
```diff
@@ -16,51 +16,44 @@
...
 
 @Singleton
-class AppConfig()
-    extends MongoConfiguration
-    with EmailConfiguration
-    with EoriCheckConfiguration
-    with CurrencyConversionConfiguration {
-  lazy val bfEmail: String = configSource("BF.email").loadOrThrow[String]
-}
+class AppConfig @Inject() (val config: Configuration)() {
```
Comment (Taiwo-AB) [inline app/uk/gov/hmrc/merchandiseinbaggage/config/AppConfig.scala:25]: Can this please be cleaned up. Can just Inject bootstrap's `ServicesConfig`. For example, something like the below:

```
@Singleton
class AppConfig @Inject() (servicesConfig: ServicesConfig, config: Configuration) {
  lazy val bfEmail: String               = config.get[String]("BF.email")
  lazy val emailUrl: String              = servicesConfig.baseUrl("email")
  lazy val eoriCheckUrl: String          = servicesConfig.baseUrl("eori-check")
  lazy val currencyConversionUrl: String = servicesConfig.baseUrl("currency-conversion")
}
```

File: app/uk/gov/hmrc/merchandiseinbaggage/connectors/CurrencyConversionConnector.scala [hunk 3]
```diff
@@ -27,13 +27,16 @@ import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class CurrencyConversionConnector @Inject() (
+  val appConfig: AppConfig,
   httpClient: HttpClient,
   @Named("currencyConversionBaseUrl") baseUrl: String
```
Comment (Taiwo-AB) [inline app/uk/gov/hmrc/merchandiseinbaggage/connectors/CurrencyConversionConnector.scala:32]: Please remove this line as it is unnecessary. Also, please remove ConfigModule [here](https://github.com/hmrc/merchandise-in-baggage/blob/DDCE-5000/conf/application.conf#L30-L35) and delete the file [here](https://github.com/hmrc/merchandise-in-baggage/blob/DDCE-5000/app/uk/gov/hmrc/merchandiseinbaggage/config/ConfigModule.scala), thanks.

File: app/uk/gov/hmrc/merchandiseinbaggage/connectors/CurrencyConversionConnector.scala [hunk 4]
```diff
@@ -27,13 +27,16 @@ import scala.concurrent.{ExecutionContext, Future}
...
-) extends CurrencyConversionConfiguration {
+) {
 
   def getConversionRate(code: String, date: LocalDate = LocalDate.now())(implicit
     hc: HeaderCarrier,
     ec: ExecutionContext
   ): Future[Seq[ConversionRatePeriod]] =
-    httpClient.GET[Seq[ConversionRatePeriod]](s"$baseUrl${currencyConversionConf.currencyConversionUrl}$date?cc=$code")
+    httpClient.GET[Seq[ConversionRatePeriod]](
+      s"$baseUrl${appConfig.currencyConversionConf.currencyConversionUrl}$date?cc=$code"
```
Comment (Taiwo-AB) [inline app/uk/gov/hmrc/merchandiseinbaggage/connectors/CurrencyConversionConnector.scala:40]: After cleanup, can update the url to `s"${appConfig.currencyConversionUrl}/currency-conversion/rates/$date?cc=$code"`, thanks

File: app/uk/gov/hmrc/merchandiseinbaggage/connectors/EmailConnector.scala [hunk 5]
```diff
@@ -30,12 +30,11 @@ trait EmailConnector {
   def sendEmails(emailInformation: DeclarationEmailInfo)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Int]
 }
 
-class EmailConnectorImpl @Inject() (http: HttpClient, @Named("emailBaseUrl") baseUrl: String)
-    extends EmailConnector
-    with EmailConfiguration {
+class EmailConnectorImpl @Inject() (config: AppConfig, http: HttpClient, @Named("emailBaseUrl") baseUrl: String)
```
Comment (Taiwo-AB) [inline app/uk/gov/hmrc/merchandiseinbaggage/connectors/EmailConnector.scala:33]: Same here any reference to `ConfigModule` will be removed, please remove the `Named` stuff (`@Named("emailBaseUrl") baseUrl: String`) and also its import, thanks.

Comment (Taiwo-AB) [inline app/uk/gov/hmrc/merchandiseinbaggage/connectors/EmailConnector.scala:33]: use `appConfig` instead of `config` for consistency with the `CurrencyConversionConnector`, thanks.

File: app/uk/gov/hmrc/merchandiseinbaggage/connectors/EoriCheckConnector.scala [hunk 6]
```diff
@@ -19,16 +19,19 @@ package uk.gov.hmrc.merchandiseinbaggage.connectors
...
 import uk.gov.hmrc.merchandiseinbaggage.model.api.Eori
 import uk.gov.hmrc.merchandiseinbaggage.model.api.checkeori.CheckResponse
 
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class EoriCheckConnector @Inject() (httpClient: HttpClient, @Named("eoriCheckBaseUrl") base: String)
-    extends EoriCheckConfiguration {
+class EoriCheckConnector @Inject() (
+  config: AppConfig,
```
Comment (Taiwo-AB) [inline app/uk/gov/hmrc/merchandiseinbaggage/connectors/EoriCheckConnector.scala:30]: use `appConfig` instead of config for consistency with the `CurrencyConversionConnector`, thanks.

File: app/uk/gov/hmrc/merchandiseinbaggage/connectors/EoriCheckConnector.scala [hunk 7]
```diff
@@ -19,16 +19,19 @@ package uk.gov.hmrc.merchandiseinbaggage.connectors
...
+  httpClient: HttpClient,
+  @Named("eoriCheckBaseUrl") base: String
```
Comment (Taiwo-AB) [inline app/uk/gov/hmrc/merchandiseinbaggage/connectors/EoriCheckConnector.scala:32]: Same here as any reference to `ConfigModule` will be removed, please remove the `Named` stuff (`@Named("eoriCheckBaseUrl") baseUrl: String`) and also its import, thanks.

File: app/uk/gov/hmrc/merchandiseinbaggage/connectors/EoriCheckConnector.scala [hunk 8]
```diff
@@ -19,16 +19,19 @@ package uk.gov.hmrc.merchandiseinbaggage.connectors
...
+) {
 
   def checkEori(eori: Eori)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[List[CheckResponse]] =
-    httpClient.GET[List[CheckResponse]](s"$base${eoriCheckConf.eoriCheckUrl}${eori.toString}")
+    httpClient.GET[List[CheckResponse]](s"$base${config.eoriCheckConf.eoriCheckUrl}${eori.toString}")
```
Comment (Taiwo-AB) [inline app/uk/gov/hmrc/merchandiseinbaggage/connectors/EoriCheckConnector.scala:36]: Please update url to `s"${appConfig.eoriCheckUrl}/check-eori-number/check-eori/${eori.toString}"` after cleanup has been done, thanks

File: test/uk/gov/hmrc/merchandiseinbaggage/stubs/CurrencyConversionStub.scala [hunk 9]
```diff
@@ -20,22 +20,22 @@ import java.time.LocalDate.now
...
+import com.google.inject.Inject
 import play.api.http.Status.OK
 import play.api.libs.json.Json
 import uk.gov.hmrc.merchandiseinbaggage.CoreTestData
-import uk.gov.hmrc.merchandiseinbaggage.config.CurrencyConversionConfiguration
+import uk.gov.hmrc.merchandiseinbaggage.config.AppConfig
 import uk.gov.hmrc.merchandiseinbaggage.model.api.ConversionRatePeriod
 
-object CurrencyConversionStub extends CurrencyConversionConfiguration with CoreTestData {
+class CurrencyConversionStub @Inject() (config: AppConfig) extends CoreTestData {
```
Comment (Taiwo-AB) [inline test/uk/gov/hmrc/merchandiseinbaggage/stubs/CurrencyConversionStub.scala:30]: No need to extend `CoreTestData` and inject `AppConfig`. This can be just be `object CurrencyConversionStub {`. Please update, thanks

File: test/uk/gov/hmrc/merchandiseinbaggage/stubs/CurrencyConversionStub.scala [hunk 10]
```diff
@@ -20,22 +20,22 @@ import java.time.LocalDate.now
...
 
   def givenCurrencyConversion(code: String = "EUR")(implicit server: WireMockServer): StubMapping =
     server.stubFor(
-      get(urlMatching(s"${currencyConversionConf.currencyConversionUrl}(.*)?cc=$code"))
+      get(urlMatching(s"${config.currencyConversionConf.currencyConversionUrl}(.*)?cc=$code"))
```
Comment (Taiwo-AB) [inline test/uk/gov/hmrc/merchandiseinbaggage/stubs/CurrencyConversionStub.scala:34]: This can just be `get(urlMatching(s"/currency-conversion/rates/(.*)?cc=$code"))`

File: test/uk/gov/hmrc/merchandiseinbaggage/stubs/EmailStub.scala [hunk 11]
```diff
@@ -20,10 +20,9 @@ import com.github.tomakehurst.wiremock.WireMockServer
 import com.github.tomakehurst.wiremock.client.WireMock.{aResponse, equalToJson, post, urlPathEqualTo}
 import com.github.tomakehurst.wiremock.stubbing.StubMapping
 import play.api.libs.json.Json
-import uk.gov.hmrc.merchandiseinbaggage.config.EmailConfiguration
 import uk.gov.hmrc.merchandiseinbaggage.model.DeclarationEmailInfo
 
-object EmailStub extends EmailConfiguration {
+object EmailStub {
 
   def givenEmailSuccess(declarationEmailInfo: DeclarationEmailInfo)(implicit server: WireMockServer): StubMapping =
```
Comment (Taiwo-AB) [inline test/uk/gov/hmrc/merchandiseinbaggage/stubs/EmailStub.scala:27]: Please use play.api.http.Status.OK and fix the scalastyle warning, thanks

File: test/uk/gov/hmrc/merchandiseinbaggage/stubs/EoriCheckStub.scala [hunk 12]
```diff
@@ -19,17 +19,18 @@ package uk.gov.hmrc.merchandiseinbaggage.stubs
...
+import com.google.inject.Inject
 import play.api.libs.json.Json
 import uk.gov.hmrc.merchandiseinbaggage.CoreTestData
-import uk.gov.hmrc.merchandiseinbaggage.config.EoriCheckConfiguration
+import uk.gov.hmrc.merchandiseinbaggage.config.AppConfig
 import uk.gov.hmrc.merchandiseinbaggage.model.api.Eori
 import uk.gov.hmrc.merchandiseinbaggage.model.api.checkeori.CheckResponse
 
-object EoriCheckStub extends EoriCheckConfiguration with CoreTestData {
+class EoriCheckStub @Inject() (config: AppConfig) extends CoreTestData {
```
Comment (Taiwo-AB) [inline test/uk/gov/hmrc/merchandiseinbaggage/stubs/EoriCheckStub.scala:29]: No need to inject `AppConfig`. This can be just be `object EoriCheckStub extends CoreTestData {`. Please update and remove unused imports after cleanup, thanks.

File: test/uk/gov/hmrc/merchandiseinbaggage/stubs/EoriCheckStub.scala [hunk 13]
```diff
@@ -19,17 +19,18 @@ package uk.gov.hmrc.merchandiseinbaggage.stubs
...
 
   def givenEoriCheck(eori: Eori, checkResponse: List[CheckResponse])(implicit server: WireMockServer): StubMapping =
     server.stubFor(
-      get(urlPathEqualTo(s"${eoriCheckConf.eoriCheckUrl}${eori.toString}"))
+      get(urlPathEqualTo(s"${config.eoriCheckConf.eoriCheckUrl}${eori.toString}"))
```
Comment (Taiwo-AB) [inline test/uk/gov/hmrc/merchandiseinbaggage/stubs/EoriCheckStub.scala:33]: This can just be `get(urlPathEqualTo(s"/check-eori-number/check-eori/${eori.toString}"))`

File: test/uk/gov/hmrc/merchandiseinbaggage/connectors/CurrencyConversionConnectorSpec.scala [hunk 14]
```diff
@@ -24,9 +24,10 @@ import java.time.LocalDate
```
Comment (Taiwo-AB) [inline test/uk/gov/hmrc/merchandiseinbaggage/connectors/CurrencyConversionConnectorSpec.scala:24]: Please revert these changes as `CurrencyConversionStub` doesn't need to a class, thanks.

File: test/uk/gov/hmrc/merchandiseinbaggage/stubs/EoriCheckStub.scala [hunk 15]
```diff
@@ -19,17 +19,18 @@ package uk.gov.hmrc.merchandiseinbaggage.stubs
...
         .willReturn(
           aResponse()
             .withStatus(200)
```
Comment (Taiwo-AB) [inline test/uk/gov/hmrc/merchandiseinbaggage/stubs/EoriCheckStub.scala:36]: Please use play.api.http.Status.OK and fix the scalastyle warning, thanks

File: test/uk/gov/hmrc/merchandiseinbaggage/connectors/EoriCheckConnectorSpec.scala [hunk 16]
```diff
@@ -20,25 +20,26 @@ import play.api.libs.json.Json
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.merchandiseinbaggage.model.api.Eori
 import uk.gov.hmrc.merchandiseinbaggage.model.api.checkeori.CheckResponse
-import uk.gov.hmrc.merchandiseinbaggage.stubs.EoriCheckStub._
+import uk.gov.hmrc.merchandiseinbaggage.stubs.EoriCheckStub
 import uk.gov.hmrc.merchandiseinbaggage.{BaseSpecWithApplication, WireMock}
```
Comment (Taiwo-AB) [inline test/uk/gov/hmrc/merchandiseinbaggage/connectors/EoriCheckConnectorSpec.scala:25]: Please revert these changes as `EoriCheckStub` doesn't need to a class, thanks.

File: test/uk/gov/hmrc/merchandiseinbaggage/pact/VerifyContractSpec.scala [hunk 17]
```diff
@@ -29,7 +29,9 @@ import scala.concurrent.duration._
 
 class VerifyContractSpec extends PactVerifySuite with CoreTestData {
 
-  private val pactDir = scala.util.Properties.envOrElse("PACTTEST", "../merchandise-in-baggage-frontend/pact/")
+  private val pactDir                = scala.util.Properties.envOrElse("PACTTEST", "../merchandise-in-baggage-frontend/pact/")
+  private val currencyConversionStub = app.injector.instanceOf[CurrencyConversionStub]
```
Comment (Taiwo-AB) [inline test/uk/gov/hmrc/merchandiseinbaggage/pact/VerifyContractSpec.scala:33]: Please revert these changes as `EoriCheckStub` and `CurrencyConversionStub` don't need to be classes, thanks.

File: test/uk/gov/hmrc/merchandiseinbaggage/config/AppConfigSpec.scala [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.merchandiseinbaggage.config
+
+import org.scalatest.matchers.must.Matchers
+import uk.gov.hmrc.merchandiseinbaggage.{BaseSpecWithApplication, WireMock}
+import uk.gov.hmrc.merchandiseinbaggage.config.AppConfig.{CurrencyConversionConf, EmailConf, EoriCheckConf}
+
+class AppConfigSpec extends BaseSpecWithApplication with Matchers {
```
Comment (Taiwo-AB) [inline test/uk/gov/hmrc/merchandiseinbaggage/config/AppConfigSpec.scala:23]: This would need rewriting once `AppConfig` has been cleaned up. Also, please remove `Matchers` as it is already brought in with `BaseSpecWithApplication`, thanks

File: app/uk/gov/hmrc/merchandiseinbaggage/connectors/CurrencyConversionConnector.scala (hunk 3 above)
Comment (gastonracui) [reply app/uk/gov/hmrc/merchandiseinbaggage/connectors/CurrencyConversionConnector.scala:32]: ohh yeah I missed removing ConfigModule, thanks Taiwo! do I need to also remove the currency-conversion configs (protocol, hots, port) from application.conf or is it used for something else?

Comment (Taiwo-AB) [reply app/uk/gov/hmrc/merchandiseinbaggage/connectors/CurrencyConversionConnector.scala:32]: No worries. No, don't remove those ones. `ServicesConfig` in bootstrap uses it to build the url.

Comment (Taiwo-AB) [reply app/uk/gov/hmrc/merchandiseinbaggage/connectors/CurrencyConversionConnector.scala:32]: If you look into `servicesConfig.baseUrl` you will see this :)

File: app/uk/gov/hmrc/merchandiseinbaggage/config/AppConfig.scala (hunk 2 above)
Comment (Taiwo-AB) [reply app/uk/gov/hmrc/merchandiseinbaggage/config/AppConfig.scala:25]: Also if you do not want to inject configuration you can update bfEmail to `servicesConfig.getString("BF.email")`

**Review by Taiwo-AB — APPROVED** [review]
